File: 2831-find-the-longest-equal-subarray/2831-find-the-longest-equal-subarray.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestEqualSubarray = function(nums, k) {
    const map = {}
    let l = 0, res = 0
    for (let r=0; r < nums.length; r++) {
        map[nums[r]] = map[nums[r]] + 1 || 1
        res = Math.max(res, map[nums[r]])
        if (r-l+1-res > k) {
            map[nums[l++]]--
        }
    }
    return res
};

File: 2831-find-the-longest-equal-subarray/2831-find-the-longest-equal-subarray.py
class Solution:
    def longestEqualSubarray(self, nums: List[int], k: int) -> int:
        l, count, maxf = 0, defaultdict(int), 0
        for r, n in enumerate(nums):
            count[n] = count.get(n, 0) + 1
            maxf = max(maxf, count[n])
            if r-l+1-maxf > k:
                count[nums[l]] -= 1
                l += 1
        return maxf